fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/prometheus.rb

```diff
@@ -0,0 +1,296 @@
+require 'prometheus/client'
+require 'prometheus/client/formats/text'
+require 'fluent/plugin/prometheus/placeholder_expander'
+
+module Fluent
+  module Plugin
+    module PrometheusLabelParser
+      def configure(conf)
+        super
+        # Check if running with multiple workers
+        sysconf = if self.respond_to?(:owner) && owner.respond_to?(:system_config)
+          owner.system_config
+        elsif self.respond_to?(:system_config)
+          self.system_config
+        else
+          nil
+        end
+        @multi_worker = sysconf && sysconf.workers ? (sysconf.workers > 1) : false
+      end
+
+      def parse_labels_elements(conf)
+        base_labels = Fluent::Plugin::Prometheus.parse_labels_elements(conf)
+
+        if @multi_worker
+          base_labels[:worker_id] = fluentd_worker_id.to_s
+        end
+
+        base_labels
+      end
+    end
+
+    module Prometheus
+      class AlreadyRegisteredError < StandardError; end
+
+      def self.parse_labels_elements(conf)
+        labels = conf.elements.select { |e| e.name == 'labels' }
+        if labels.size > 1
+          raise ConfigError, "labels section must have at most 1"
+        end
+
+        base_labels = {}
+        unless labels.empty?
+          labels.first.each do |key, value|
+            labels.first.has_key?(key)
+
+            # use RecordAccessor only for $. and $[ syntax
+            # otherwise use the value as is or expand the value by RecordTransformer for ${} syntax
+            if value.start_with?('$.') || value.start_with?('$[')
+              base_labels[key.to_sym] = PluginHelper::RecordAccessor::Accessor.new(value)
+            else
+              base_labels[key.to_sym] = value
+            end
+          end
+        end
+
+        base_labels
+      end
+
+      def self.parse_metrics_elements(conf, registry, labels = {})
+        metrics = []
+        conf.elements.select { |element|
```

Comment:
prefer using "do/end" for multi-line syntax

Reply:
removed this copy of prometheus.rb as  dependency is sorted out and will be loaded from installed path of fluent-plugin-prometheus 1.8.5 gem.